src/Adapters/Storage/Mongo/MongoCollection.js

```diff
@@ -101,14 +143,21 @@ export default class MongoCollection {
     return countOperation;
   }
 
-  distinct(field, query) {
+  distinct(field, query, hint) {
+    if (hint) {
+      return this._mongoCollection.distinct(field, query).hint;
+    }
+
     return this._mongoCollection.distinct(field, query);
   }
 
-  aggregate(pipeline, { maxTimeMS, readPreference } = {}) {
-    return this._mongoCollection
-      .aggregate(pipeline, { maxTimeMS, readPreference })
-      .toArray();
+  aggregate(pipeline, { maxTimeMS, readPreference, hint } = {}) {
+    if (hint) {
```

Comment:
You forgot to add an else statement here.

Reply:
Updated... good catch!

Reply:
You're saying to use the rest API in the tests?

Reply:
Yes here a few test examples.

https://github.com/parse-community/parse-server/blob/master/spec/ParseQuery.FullTextSearch.spec.js#L66

https://github.com/parse-community/parse-server/blob/master/spec/ParsePolygon.spec.js#L351

I'm curious to see what will be returned. This will reflect on how the sdks will work.

Reply:
Can you check the Travis CI build? https://travis-ci.org/parse-community/parse-server/builds/634052278?utm_source=github_status&utm_medium=notification

It says your code needs linting.

Reply:
I'm having trouble with understanding how these tests work... I cannot get them all to pass when running them by pulling master without any changes... also I see where some tests in the Travis CI are failing but I'm having a hard time understanding where to go to fix them.

Reply:
Its a linter issue but specifically flow typing. Since you added hint you have to add what type hint is to the functions.

<img width="795" alt="Screen Shot 2020-01-07 at 11 35 34 PM" src="https://user-images.githubusercontent.com/9830365/71953047-8ee62100-31a6-11ea-8773-0ff4ab99d242.png">